### PR TITLE
MODORDSTOR-132 - Title schema updates

### PR DESCRIPTION
**Pull request by khandramai**
[MODORDSTOR-132](https://issues.folio.org/browse/MODORDSTOR-132) - Title schema updates

## Purpose
We need several more optional fields in the title record.

## Approach
New non-required fields added to title schema, samples were updated.

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Examples exist for all schemas
  - [x] Descriptions exist for all schema properties
  - [x] All schemas pass raml-cop linting
- Does this introduce breaking changes?
  - [ ] Were there any schema changes?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: mod-orders-storage/schemas/title.json [hunk 1]
```diff
@@ -26,6 +26,42 @@
...
+    "publisher": {
+      "description": "Publisher of the material",
+      "type": "string"
+    },
+    "edition": {
+      "description": "Edition of the material",
+      "type": "string"
+    },
+    "publishedDate": {
+      "description": "Edition of the material",
```
Comment (KVupp) [inline mod-orders-storage/schemas/title.json]: description needs to be changes, and type must be date if it is publishedDate?

Comment (KVupp) [reply mod-orders-storage/schemas/title.json]: oh I see the type needn't changed, we just expect it to be a year. just the description then

Comment (KVupp) [reply mod-orders-storage/schemas/title.json]: Also the field is named "publicationDate" in the story, might want to correct that as well

Comment (khandramai) [reply mod-orders-storage/schemas/title.json]: @craigmcnally , could you please clarify:
according to story description this field names "Publication Date", but according wiki page - "publishedDate". Is this OK to left it as is, i.e. "publishedDate"?

Comment (craigmcnally) [reply mod-orders-storage/schemas/title.json]: Either way is fine.  If there's a strong preference to how this is displayed in the UI, we can use whatever label desired in the UI.   As long as the description is clear in the schema, the name of the field is of little consequence.

**Pull request merged**